store/store.go

```diff
@@ -0,0 +1,119 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+)
+
+type Store interface {
+	Get(model.Fingerprint) (*types.Alert, error)
+	Set(*types.Alert) error
+	Delete(model.Fingerprint) error
+	List() <-chan *types.Alert
+	SetGCCallback(func(*types.Alert))
+}
+
+var (
+	ErrNotFound = errors.New("alert not found")
+)
+
+type Alerts struct {
+	sync.Mutex
+	c  map[model.Fingerprint]*types.Alert
```

Comment:
What does `c` stand for? `cache`?

Reply:
yep! happy to rename, just seemed like the Alerts struct has a small enough responsibility that it didn't really matter.

Reply:
Right, cool to keep structs that small. How about a small comment for _future-me_?